src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateVersions = [];
+  let currentState = state;
```

Comment:
You are assigning the reference of state object to the currentState. Any modifications on currentState will also affect the original state object. Use spread syntax or Object.assign to create a shallow copy of the state object.